package-res/resources/web/pentaho/lang/_doc/Event.jsdoc

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @classDesc The `Event` class is the base class of event objects emitted by an
+ * [EventSource]{@link pentaho.lang.EventSource}.
+ *
+ * The source of an event is the object that emits it and
+ * is given by its [source]{@link pentaho.lang.Event#source} property.
+ *
+ * ##### Event Type
+ *
+ * Events have a _type_ that is given by its [type]{@link pentaho.lang.Event#type} property.
+ * The type of an event is the _id_ used to register to it,
+ * using [EventSource#on]{@link pentaho.lang.EventSource#on}.
+ *
+ * An event type corresponds to a single subclass of `Event`.
+ * However, an `Event` subclass can be used by several event types.
+ * The `Event` subclass of an event type mainly depends on the data that it is associated with.
+ *
+ * ##### Event Cancellation
+ *
+ * Certain types of events are used to signal that an _action_ is about to execute
+ * (or that a phase of an already executing action is about to start).
+ * When the execution of the action can be canceled by the event listeners,
+ * the event is said to be _cancelable_.
+ * That characteristic is exposed by the [isCancelable]{@link pentaho.lang.Event#isCancelable} property.
+ *
+ * When an event is canceled, not only its corresponding action is canceled,
+ * but the listeners of unprocessed registrations are not notified.
```

Comment:
When an event is canceled, its corresponding action is also canceled and the listeners of unprocessed registrations are not notified.